src/mon/MDSMonitor.cc

```diff
@@ -767,6 +768,19 @@ bool MDSMonitor::preprocess_command(MMonCommand *m)
       if (p != &mdsmap)
 	delete p;
     }
+  } else if (prefix == "mds metadata") {
+    mds_gid_t gid = MDS_GID_NONE;
+    if (!cmd_getval(g_ceph_context, cmdmap, "gid", gid)) {
```

Comment:
Users rarely want to refer to their daemons by GID.  Ideally, this would do the same trick as MDSMonitor::fail_mds, to take a parameter that can be an ID, GID or rank.